astropy/time/core.py

```diff
@@ -582,10 +582,10 @@ def _get_time_fmt(
         """
         if format is None:
             # If val and val2 broadcasted shape is (0,) (i.e. empty array input) then we
-            # cannot guess format from the input values.  Instead use the default
```

Comment:
Is that really important though? The internal representation is always the same. So why not simply always create an empty time instance for empty input?

Reply:
I guess the question then is what the format of the empty time instance should be.

Also, don't forget this line is also for all-masked values.

Reply:
> I guess the question then is what the format of the empty time instance should be.

That's always confusing me. Why are things like "format" and "precision" instance properties? The internal representation is always the same. It's affecting only affecting the default repr, right?

Reply:
Oops, this change is actually introducing a new regression since Quantity is only allowed for `TimeDelta` not `Time`.
```
In [3]: Time([] * u.s)
Out[3]: <Time object: scale='utc' format='ymdhms' value=[]>
```
Instead for `Time` this should raise an exception like it currently does. The logic is getting ugly, but I think it could look like:
```
            if not (isinstance(self, TimeDelta) and isinstance(val, u.Quantity)) and (
                empty_array or np.all(mask)
            ):
```

Reply:
> I guess the question then is what the format of the empty time instance should be.

For `TimeDelta` the default format of `jd` is fine. As mentioned for `Time` we don't have to worry about it since an empty input with no format should always raise an exception.

Reply:
@taldcroft - OK, I pushed your change

@maxnoe - I agree that it is a little strange to error for what indeed is in a sense a change in `repr` - though this affects things like `.value` as well. But it is a bit out of scope here - I just want to fix a regression!  Also, as noted earlier, the same logic is applied for fully masked input.

Reply:
@mhvk's point about `.value` is crucial. Without a defined `.format` then `.value` is undefined. The `.value` is a primary functional output of the `Time/TimeDelta` classes.

Reply:
And `format` also tells the class initializer and the object how to interpret input values. E.g.
```
tm = Time(["2023-01-01"])

tm[0] = 100.0
# ValueError

tm.format = "unix"
tm[0] = 100.0
# OK
```